exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -110,6 +110,328 @@ type KubeletConfig struct {
+	TransparentHugePageMadvise TransparentHugePage = "madvise"
+
+	// TransparentHugePageEnabledNever ...
+	TransparentHugePageNever TransparentHugePage = "never"
+
+	// TransparentHugePageDefer ...
+	TransparentHugePageDefer TransparentHugePage = "defer"
+
+	// TransparentHugePageDeferMadvise ...
+	TransparentHugePageDeferMadvise TransparentHugePage = "defer+madvise"
+)
+
+// VirtualMemoryConfig specifies the fields for tuning virtual memory subsystem of the Linux kernel and the writeout of dirty data to disk.
+type VirtualMemoryConfig struct {
+	// MaxMapCount specifies the maximum number of memory map areas a process may have.
+	// +kubuilder:validation:Minimum=65530
+	// +kubuilder:validation:Maximum=262144
+	// +kubuilder:default=65530
+	// +optional
+	MaxMapCount *uint32 `json:"maxMapCount,omitempty"`
+
+	// VFSCachePressure specifies the percentage value that controls tendency of the kernel to reclaim the memory, which is used for caching of directory and inode objects.
+	// +kubuilder:validation:Minimum=1
+	// +kubuilder:validation:Maximum=500
+	// +kubuilder:default=100
+	// +optional
+	VFSCachePressure *uint16 `json:"vfsCachePressure,omitempty"`
+
+	// Swappiness specifies aggressiveness of the kernel in swapping memory pages.
+	// Higher values will increase aggressiveness, lower values decrease the amount of swap.
+	// +kubuilder:validation:Minimum=1
+	// +kubuilder:validation:Maximum=100
+	// +kubuilder:default=60
+	// +optional
+	Swappiness *uint8 `json:"swappiness,omitempty"`
+
+	// TODO: Upper limit of the swapFileSize should be verified by the reconciler.
+	// SwapFileSizeMB specifies size in MB of a swap file will be created on the agent nodes from this node pool.
+	// Max value of SwapFileSizeMB should be the size of temporary disk(/dev/sdb). Refer: https://learn.microsoft.com/en-us/azure/virtual-machines/managed-disks-overview#temporary-disk
+	// +kubuilder:validation:Minimum=1
+	// +optional
+	SwapFileSizeMB *uint16 `json:"swapFileSizeMB,omitempty"`
+
+	// TransparentHugePageEnabled specifies various modes of Transparent Hugepages. Refer to https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html#admin-guide-transhuge for more details.
+	// +kubebuilder:validation:Enum=always;madvise;never
+	// +kubuilder:default=always
+	// +optional
+	TransparentHugePageEnabled *TransparentHugePage `json:"transparentHugePageEnabled,omitempty"`
+
+	// TransparentHugePageDefrag specifies whether the kernel should make aggressive use of memory compaction to make more hugepages available.
+	// Refer to https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html#admin-guide-transhuge for more details.
+	// +kubebuilder:validation:Enum=always;defer;defer+madvise;madvise;never
+	// +kubuilder:default=madvise
+	// +optional
+	TransparentHugePageDefrag *TransparentHugePage `json:"TransparentHugePageDefrag,omitempty"`
+}
+
+// LinuxOSConfig specifies the custom linux OS settings and configurations
+type LinuxOSConfig struct {
```

Comment:
Hey there, great work here so far. Let me ask you one thing: what do you think about using containerservice.LinuxOSConfig as the model and have the exact same struct here? This way the implementation will be consistent with the Azure API, the code will be much simpler and less convoluted in order to convert the types and also consistent with how custom Kubelet config was implemented?

Just wondering maybe if this API was already defined and agreed with core maintainers?

Reply:
Thanks for a prompt course correction comment @juanmaia! Makes sense on defining this API as close as possible to Azure API. 
I also checked with CAPZ core maintainers, they agree with your suggestion too.

Reply:
Addressed via 5bac843dfe5781b635e6bcd88f873b131475102b. Please take a look!